src/controllers/__tests__/userDataProcessingController.test.ts

```diff
@@ -0,0 +1,259 @@
+import UserDataProcessingService from "src/services/userDataProcessingService";
+import { EmailAddress } from "../../../generated/backend/EmailAddress";
+import { FiscalCode } from "../../../generated/backend/FiscalCode";
+import { SpidLevelEnum } from "../../../generated/backend/SpidLevel";
+import mockReq from "../../__mocks__/request";
+import mockRes from "../../__mocks__/response";
+import ApiClient from "../../services/apiClientFactory";
+import { SessionToken, WalletToken } from "../../types/token";
+import { User } from "../../types/user";
+import UserDataProcessingController from "../userDataProcessingController";
+
+const aTimestamp = 1518010929530;
+const aUserDataProcessingResponse = {
+  status: 200,
+  value: {
+    _etag: "bdb8f644-132c-4f3c-a051-5887fc8058b1",
+    _rid: "AAAAAQAAAAgAAAAAAAAAAQ==",
+    _self: "/dbs/AAAAAQ==/colls/AAAAAQAAAAg=/docs/AAAAAQAAAAgAAAAAAAAAAQ==/",
+    _ts: 1582553174,
+    choice: "DOWNLOAD",
+    createdAt: "2020-02-24T14:06:14.513Z",
+    fiscalCode: "SPNDNL80A13Y555X",
+    id: "SPNDNL80A13Y555X-DOWNLOAD-0000000000000000",
+    status: "PENDING",
+    userDataProcessingId: "SPNDNL80A13Y555X-DOWNLOAD",
+    version: 0
+  }
+};
+
+const aFiscalNumber = "GRBGPP87L04L741X" as FiscalCode;
+const anEmailAddress = "garibaldi@example.com" as EmailAddress;
+const aValidName = "Giuseppe Maria";
+const aValidFamilyname = "Garibaldi";
+const aValidSpidLevel = SpidLevelEnum["https://www.spid.gov.it/SpidL2"];
+
+const userDataProcessingMissingErrorResponse = ResponseErrorInternal(
+  "Not Found"
+);
+
+// mock for a valid User
+const mockedUser: User = {
+  created_at: aTimestamp,
+  family_name: aValidFamilyname,
+  fiscal_code: aFiscalNumber,
+  name: aValidName,
+  session_token: "123hexToken" as SessionToken,
+  spid_email: anEmailAddress,
+  spid_level: aValidSpidLevel,
+  spid_mobile_phone: "3222222222222" as NonEmptyString,
+  wallet_token: "123hexToken" as WalletToken
+};
+
+const mockedUserDataProcessingChoice: UserDataProcessingChoice =
+  UserDataProcessingChoiceEnum.DOWNLOAD;
+
+const mockedUserDataProcessingChoiceRequest: UserDataProcessingChoiceRequest = {
+  choice: mockedUserDataProcessingChoice
+};
+
+const badRequestErrorResponse = {
+  detail: expect.any(String),
+  status: 400,
+  title: expect.any(String),
+  type: undefined
+};
+
+const mockGetUserDataProcessing = jest.fn();
+const mockUpsertUserDataProcessing = jest.fn();
+jest.mock("../../services/userDataProcessingService", () => {
+  return {
+    default: jest.fn().mockImplementation(() => ({
+      getUserDataProcessing: mockGetUserDataProcessing,
+      upsertUserDataProcessing: mockUpsertUserDataProcessing
+    }))
+  };
+});
+
+describe("UserDataProcessingController#getUserDataProcessing", () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it("calls the getUserDataProcessing on the UserDataProcessingService with valid values", async () => {
```

Comment:
test descriptions must begin with "should .." and describe the expectation. ie.

it("should return a successful response in case a valid payload is provided")

same for every other description.